### DISCUSSION TITLE
请问如何使用MiraiGo生成的devise.json？

### DISCUSSION THREAD

**Discussion by Sora233** [mamoe/mirai#760] [Q&A] [answered]
目前使用Miraigo可以正常登陆。
我使用的是官方mirai-console-loader，版本是2.0-M1-1，生成的devise.json比miraigo的要多几项，我把miraigo的内容按字节组的格式填进去，包括imei,finger_print,boot_id以及其他的，但有一些剩余字段没法获取，登陆的时候依旧提示上网环境异常/常用地登陆。

**Comment by Karlatemp** [comment]
对照 https://github.com/mamoe/mirai/blob/7015501f535845b3dfc18732091dd30b131239a8/mirai-core-api/src/commonMain/kotlin/utils/DeviceInfo.kt#L69-L95

**Comment by ryoii** [comment, marked as answer]
如果清楚剩余字段是什么内容，可以尝试从这里生成 https://ryoii.github.io/mirai-devicejs-generator/
